lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/IO.hs

```diff
@@ -193,6 +195,19 @@ createAddress c w =
             let (r,s1) = Wallet.createAddress c s0
             in  (Delta.Replace s1, r)
 
+data WalletPublicIdentity = WalletPublicIdentity
+    { pubXpub :: XPub
+    , pubNextUser :: Word31
+    }
+    deriving Show
```

Comment:
Could you move this type to `Cardano.Wallet.Deposit.Pure` and add a function `getWalletPublicIdentity`? I think this concept stands on its own to merit inclusion in the pure Haskell API.